tests/dataset_builders/pie/test_lisa.py

```diff
@@ -0,0 +1,395 @@
+from pytorch_ie.documents import TextDocumentWithLabeledSpansAndBinaryRelations, TokenBasedDocument
+from transformers import AutoTokenizer, PreTrainedTokenizer
+
+from dataset_builders.pie.lisa.lisa import Lisa
+from tests.dataset_builders import PIE_BASE_PATH
+
+
+@dataclasses.dataclass
+class TestTokenDocumentWithLabeledSpans(TokenBasedDocument):
+    labeled_spans: AnnotationList[LabeledSpan] = annotation_field(target="tokens")
+
+
+@dataclasses.dataclass
+class TestTokenDocumentWithLabeledSpansAndBinaryRelations(TestTokenDocumentWithLabeledSpans):
+    binary_relations: AnnotationList[BinaryRelation] = annotation_field(target="labeled_spans")
+
+
+disable_caching()
+
+DATASET_NAME = "lisa"
+PIE_DATASET_PATH = PIE_BASE_PATH / DATASET_NAME
+SPLIT_SIZES = {"train": 118}
+
+
+@pytest.fixture(scope="module", params=["default", "merge_fragmented_spans"])
+def dataset_variant(request) -> str:
+    return request.param
+
+
+@pytest.fixture(scope="module")
+def dataset(dataset_variant) -> DatasetDict:
+    return DatasetDict.load_dataset(str(PIE_DATASET_PATH), name=dataset_variant)
+
+
+def test_dataset(dataset):
+    assert dataset is not None
+    assert {name: len(ds) for name, ds in dataset.items()} == SPLIT_SIZES
+
+
+@pytest.fixture(scope="module")
+def document(dataset) -> Union[BratDocument, BratDocumentWithMergedSpans]:
+    return dataset["train"][0]
+
+
+def test_document(document, dataset_variant):
+    assert document is not None
+    assert document.text.startswith(
+        "Hallo liebe <user>, danke für deine lieben Worte, hier im Forum sind alle so nett und hilfsbereit."
+    )
+    if dataset_variant == "default":
+        assert isinstance(document, BratDocument)
+        span = document.spans[0]
+        assert isinstance(span, LabeledMultiSpan)
+        assert len(document.spans) == 10
+        # sort the entities by their start position and convert them to tuples
+        # check the first ten entities after sorted
+        sorted_entity_tuples = [
+            (ent.target[ent.slices[0][0] : ent.slices[0][1]], ent.label)
+            for ent in sorted(document.spans, key=lambda ent: ent.slices[0][0])
+        ]
+        # Checking the first ten entities
+        assert sorted_entity_tuples == [
+            ("Brennen", "DISORDER"),
+            ("massive Nervosität", "DISORDER"),
+            ("Citalopram", "DRUG"),
+            ("Angstpatienten", "DISORDER"),
+            ("AD", "DRUG"),
+            ("Citalopram", "DRUG"),
+            ("nicht mehr nehme", "CHANGE_TRIGGER"),
+            ("gehts mir wieder viel besser", "OPINION"),
+            ("schreckliche Überdrehtheit", "DISORDER"),
+            ("dachte echt ich werde verrückt", "OPINION"),
+        ]
+
+        # check the relations
+        assert len(document.relations) == 6
+        # check the first ten relations
+        relation_tuples = [
+            (
+                rel.head.target[rel.head.slices[0][0] : rel.head.slices[0][1]],
```

Comment:
I really would create the head / tail strings from all the slices to make any fragments visible here (also below)

Reply:
Do you mean to iterate over all possible slices for head and tail?

Reply:
I meant
```suggestion
                [rel.head.target[slice_start:slice_end] for slice_start, slice_end in rel.head.slices],
```

But I just noticed, that this `sorted_entity_tuples` above also contains **only the first slice** for each multi-span... also there, you should use *all* slices (similar as in my code suggestion here)

Reply:
Exactly, that's what I was trying to ask. I was only looking at the first fragment (0th index)` rel.head.slices[0][x]` instead of iterating over all possible fragments which your suggested code does and represents in tuple format. Thanks.